pkg/webhooks/dynamicresourceallocationconfig_webhook.go

```diff
@@ -0,0 +1,119 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package webhooks
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	kueuev1alpha1 "sigs.k8s.io/kueue/apis/kueue/v1alpha1"
+	"sigs.k8s.io/kueue/pkg/features"
+)
+
+// DynamicResourceAllocationConfigWebhook validates create / update requests.
+type DynamicResourceAllocationConfigWebhook struct{}
+
+func setupWebhookForDynamicResourceAllocationConfig(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(&kueuev1alpha1.DynamicResourceAllocationConfig{}).
+		WithValidator(&DynamicResourceAllocationConfigWebhook{}).
+		Complete()
+}
+
+// +kubebuilder:webhook:path=/validate-kueue-x-k8s-io-v1alpha1-dynamicresourceallocationconfig,mutating=false,failurePolicy=fail,sideEffects=None,groups=kueue.x-k8s.io,resources=dynamicresourceallocationconfigs,verbs=create;update,versions=v1alpha1,name=vdynamicresourceallocationconfig.kb.io,admissionReviewVersions=v1
+
+var _ webhook.CustomValidator = &DynamicResourceAllocationConfigWebhook{}
+
+// ValidateCreate blocks CR creation when the feature-gate is off.
+func (w *DynamicResourceAllocationConfigWebhook) ValidateCreate(_ context.Context, obj runtime.Object) (admission.Warnings, error) {
+	if !features.Enabled(features.DynamicResourceAllocation) {
+		return nil, field.Forbidden(field.NewPath(""), "DynamicResourceAllocation feature gate is disabled")
+	}
+	draCfg := obj.(*kueuev1alpha1.DynamicResourceAllocationConfig)
+
+	var allErrs field.ErrorList
+	if draCfg.Name != "default" {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("metadata", "name"), draCfg.Name, "must be 'default'"))
+	}
+
+	if draCfg.Namespace != "kueue-system" {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("metadata", "namespace"), draCfg.Namespace, "must be 'kueue-system'"))
+	}
+
+	if len(allErrs) > 0 {
+		return nil, allErrs.ToAggregate()
+	}
+
+	// structural validation of spec
+	allErrs = append(allErrs, validateDynamicResourceAllocationConfig(draCfg)...)
+	if len(allErrs) > 0 {
+		return nil, allErrs.ToAggregate()
+	}
+	return nil, nil
+}
+
+// ValidateUpdate blocks updates when the feature-gate is off.
+func (w *DynamicResourceAllocationConfigWebhook) ValidateUpdate(_ context.Context, _ runtime.Object, newObj runtime.Object) (admission.Warnings, error) {
+	if !features.Enabled(features.DynamicResourceAllocation) {
+		return nil, field.Forbidden(field.NewPath(""), "DynamicResourceAllocation feature gate is disabled")
+	}
+	draCfg := newObj.(*kueuev1alpha1.DynamicResourceAllocationConfig)
+	var allErrs field.ErrorList
+
+	if draCfg.Name != "default" {
```

Comment:
You should move this to CEL or kubebuilder validation.

Try and limit what we put in webhook code.

Reply:
updated, the only thing in webhook now is the namespace check, which is not available at compile time and it is coming from cfg.Namespace

Reply:
I think you can remove this code. You added it as a CEL validation in the API so this is not needed anymore.